source/deploy/application-architecture.rst

```diff
@@ -0,0 +1,109 @@
+
+Application layer
+~~~~~~~~~~~~~~~~~
+
+The Mattermost Server is the heart of the platform and responsible for processing all user and system operations. It is composed of multiple modular elements:
+
+**RESTful JSON Web Service**
+Handles all incoming API requests (from web clients, apps, and integrations) and ensures that responses are formatted in JSON. Acts as the communication bridge between the clients (Access Layer) and backend systems.
+
+**Authentication client**
+Manages user authentication, ensuring secure login sessions. Integrates with traditional username/password-based authentication or enterprise-grade solutions like SSO (Single Sign-On) through Active Directory/LDAP.
+
+**Authentication Provider**
+Provides pluggable authentication frameworks to support OAuth, SSO, and third-party identity services. Particularly important for enterprise environments with centralized identity management.
+
+**Notification Service**
+
+Sends notifications through supported mediums:
+
+- Push Notifications: Real-time notifications to iOS and Android devices (via a Push Notification Service).
+- Email Notifications: Delivered to users when they are offline or need event alerts.
+
+**Data Management Service**
+Responsible for managing message data, metadata, user profiles, and logs. Ensures the integrity of data passed between the database and the server. This layer serves as the operational core of the platform, orchestrating user activities with data handling and integration capabilities.
+
+Backend Infrastructure
+~~~~~~~~~~~~~~~~~~~~~~
+
+The backend infrastructure provides the storage and data handling capabilities required for Mattermost operations. It consists of the following components:
+
+**Database Systems**
+
+Mattermost uses PostgreSQL as its primary database (supports Amazon RDS for cloud-hosted PostgreSQL) to store all persistent data, such as:
+
+- Messages
+- User accounts and credentials
+- Configuration settings
+- Team/channel metadata
+
+**File Storage**
+
+Manages all multimedia assets (e.g., file uploads, images, videos) shared across channels. Storage solutions include:
+
+- **Local Storage**: Files stored directly on the server’s filesystem.
+- **Network Attached Storage (NAS)**: Common for enterprises centralizing file storage within their network.
+- **S3**: Offers cloud-based scalable storage for larger environments or organizations with distributed deployments. The database and file storage handle scalability, ensuring efficient support for millions of messages and files while guaranteeing data consistency.
+
+**System Extensions**
+
+Mattermost is not only a collaboration tool but also a platform designed for extensibility. Key extensibility features include:
```

Comment:
Should we mention SAML and LDAP here? For most enterprises, these are keen system to integrate.